tests/test_category_checker.py

```diff
@@ -1,115 +1,187 @@
-                list_questions += categories["or"]
-            else:
-                categories["or"] = []
-            if "and" in categories:
-                list_questions += categories["and"]
-                total += len(categories["and"])
-            duplicates = []
-            for category in list_questions:
-                for category_check in data[0]["categories"]:
-                    for types in ["and", "or"]:
-                        if types in category_check:
-                            for ck in category_check[types]:
-                                if (
-                                    category_check["name"]
-                                    != categories["name"]
-                                ):
-                                    if ck["question"] == category["question"]:
-                                        intersect = intersection(
-                                            ck["options"], category["options"]
-                                        )
-                                        if len(intersect):
-                                            duplicates.append(
-                                                {
-                                                    "question": ck["question"],
-                                                    "name": category_check[
-                                                        "name"
-                                                    ],
-                                                    "option": ck["options"],
-                                                    "type": types,
-                                                }
-                                            )
-            for duplicate in duplicates:
-                for types in ["and", "or"]:
-                    dp = len(
-                        list(
-                            filter(
-                                lambda x: x["type"] == types
-                                and x["name"] == duplicate["name"]
-                                and x["question"] == duplicate["question"],
-                                duplicates,
-                            )
-                        )
-                    )
-                    if dp:
-                        if duplicate["name"] in duplicates_dict:
-                            if types in duplicates_dict[duplicate["name"]]:
-                                duplicates_dict[duplicate["name"]][types] += 1
-                            else:
-                                duplicates_dict[duplicate["name"]][types] = 1
-                            duplicates_dict[duplicate["name"]]["total"] += 1
-                        else:
-                            duplicates_dict[duplicate["name"]] = {
-                                "and": 1 if "and" == types else 0,
-                                "or": 1 if "or" == types else 0,
-                                "total": 1,
+        fake_data = [
```

Comment:
Hi @ifirmawan We don't need bunch of categories in one test, what we need is test for all the failing cases

Reply:
Also please use more readable example!

Example:
```json
{
   "name": "doesn't matter",
   "form":1,
   "categories": [{
         "name": "Category 1AB-2B",
         "and": [{
              "question": 1,
              "options": ["A","B"]
         }, {
              "question": 2,
              "options": ["B"]
         }]
    }, {
         "name": "Category 1A-2C",
         "and": [{
              "question": 1,
              "options": ["A"]
          }, {
              "question": 2,
              "options": ["B"]
         }]
    }]
}
```